.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,54 @@
+name: Clang Format Check
+on:
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - 'Tests/**'
+      - 'Samples/Common/Sources/CrashTriggers/**'
+      - '.github/workflows/clang-format.yml'
+      - '.clang-format-ignore'
+
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install clang-format-18
+      run: |
+        wget https://apt.llvm.org/llvm.sh
+        chmod +x llvm.sh
+        sudo ./llvm.sh 18
+        sudo apt-get install -y clang-format-18
+
+    - name: Define directories
+      run: echo "DIRS='Sources Tests Samples/Common/Sources/CrashTriggers' " >> $GITHUB_ENV
+
+    - name: Check formatting
+      run: |
+        find $DIRS -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' | \
+        xargs -r clang-format-18 -style=file -n -Werror
```

Comment:
By the way, maybe it's better to add back a `Makefile` with this? Or a script. As otherwise it might not be easy for contributors to fix formatting without applying the patch from PR (I would prefer to fix format before I submit changes)

Reply:
Let's do it in another PR